### ISSUE TITLE
Regression - Class list missing in Group page [1.8.16 -> 1.8.17]

### ISSUE THREAD

**Issue by kobalicek** [doxygen/doxygen#7772]
**Describe the bug**

Doxygen 1.8.17 (and 1.8.18) won't generate class list in a documentation group while 1.8.16 generates it as expected. It's worth noting that enumerations and namespaces are generated, only class list is missing.

**Attachments**

If the attachment is not working for some reason, here is the content of two files that can be used to reproduce the issue:

```c++
//! \defgroup my_group My Group
//! \brief Brief description of my group
//!
//! Detailed description of my group...

//! \addtogroup my_group
//! \{

//! I'm a namespace within \ref my_group.
namespace Namespace {}

//! I'm some enum.
enum Enum {
  //! I'm enum value
  kEnumValue = 0
};

//! I'm a class defined in \ref my_group.
class GroupedClass {
public:
  //! I'm some method in a class.
  void method(const char* str) {}

  // Undocumented
  void undocumentedMethod() {}
};
//! \}
```

Doxyfile:

```
PROJECT_NAME            = "Doxygen Test Project"
INPUT                   = .
XML_OUTPUT              = doc_xml
HTML_OUTPUT             = doc
CASE_SENSE_NAMES        = YES
RECURSIVE               = YES
QUIET                   = YES
FILE_PATTERNS           = *.h
ENABLE_PREPROCESSING    = YES
PREDEFINED              = __cplusplus=201703

WARN_IF_UNDOCUMENTED    = NO
MACRO_EXPANSION         = YES
EXTRACT_STATIC          = YES
EXTRACT_PRIVATE         = NO
EXTRACT_LOCAL_CLASSES   = NO
EXCLUDE_SYMBOLS         =

BRIEF_MEMBER_DESC       = NO
ALWAYS_DETAILED_SEC     = YES
GROUP_NESTED_COMPOUNDS  = YES
SHOW_FILES              = NO
SHOW_USED_FILES         = NO
SHOW_NAMESPACES         = NO
SHOW_INCLUDE_FILES      = NO
SOURCE_BROWSER          = NO
VERBATIM_HEADERS        = NO
REFERENCES_LINK_SOURCE  = NO
GROUP_GRAPHS            = NO
COLLABORATION_GRAPH     = NO
ENUM_VALUES_PER_LINE    = 0
SORT_BRIEF_DOCS         = NO
SORT_MEMBER_DOCS        = NO
SORT_GROUP_NAMES        = NO
DISABLE_INDEX           = YES
INLINE_INFO             = NO
SORT_BY_SCOPE_NAME      = NO
#INLINE_GROUPED_CLASSES = YES
SOURCE_TOOLTIPS         = NO
ALPHABETICAL_INDEX      = NO
HIDE_SCOPE_NAMES        = NO

CLASS_GRAPH             = NO
DIRECTORY_GRAPH         = NO

GENERATE_HTML           = YES
GENERATE_XML            = NO
GENERATE_LATEX          = NO
GENERATE_TREEVIEW       = NO
XML_PROGRAMLISTING      = NO

# Custom search support
SEARCHENGINE            = YES
SERVER_BASED_SEARCH     = YES
EXTERNAL_SEARCH         = YES
SEARCHDATA_FILE         = doc/searchdata.xml
```

[doxygen-bug.zip](https://github.com/doxygen/doxygen/files/4620724/doxygen-bug.zip)

**Comment by albert-github** [comment]
@doxygen
Looks like this is a regression due to:

> Commit: d38df57916d5902abc9dbaf5891c14898ce675da [d38df57]
> Date: Monday, October 28, 2019 9:56:03 PM
> Commit Date: Tuesday, October 29, 2019 10:54:21 PM
> Replaced QList\<Grouping\> with std::vector\<Grouping\>

as here the classes are not available anymore on the group / Module page, whilst after:

> Commit: 94e8899c8013cf99a6f90d1c8f0a7815ccc3e1c3 [94e8899]
> Date: Monday, October 28, 2019 9:29:16 PM
> Commit Date: Tuesday, October 29, 2019 10:54:17 PM
> Replaced QList\<BaseInfo\> with std::vector\<BaseInfo\>

they still are.

- See page: doc/group__my__group.html
   or
- `grep -r doc nested`

**Comment by pss959** [comment]
I just came here to report a similar bug, which probably has the same cause. I hope the extra data here helps make it easier to find the problem.
Running on this C# input:
```
/*!
 * \defgroup TestGroup Nothing to see here
 */

/*!
 * A test.
 * \ingroup TestGroup
 */
public class Class1ThatWorks {}

/*!
 * Another test.
 * \ingroup TestGroup
 */
public class Class2ThatDoesNotWork {
}
```

The second class is not added to the group if GROUP_NESTED_COMPOUNDS = YES in the Doxyfile. It is added to the group if the variable is set to NO.
Fails in Version 1.8.17; used to work in 1.8.16.

**Comment by kobalicek** [comment]
Shouldn't the commit that introduced the regression be reverted? I'm stuck with old doxygen now, because all recent versions are broken...

**Comment by albert-github** [comment]
I don't think the commit should be reverted but the problem should be solved or it should be explained why the current implementation is correct and the old one wasn't when that is the case.

**Comment by kobalicek** [comment]
I don't think not displaying classes in documentation groups should be considered a correct behavior. It's definitely a regression as older doxygen just works fine and produces the expected output. Should i upload also screenshots of this?

**Comment by albert-github** [comment]
No need for images as in the page doc/group__my__group.html it is quite clear between the 1.8.16 and newer versions (including 1.9.1)

**Comment by kobalicek** [comment]
Is there any chance that this will get fixed? Considering how many open issues are there I would suggest updating the documentation about the behavior and closing it. I will stay with 1.8.16 for now as it works well for my use cases.

**Comment by albert-github** [comment]
I did a quick test witht he 1.8.16 / 1.8.17 and the current master and it looks like the "Classes" are back but that the "Namespace" is gone now.

**Comment by doxygen** [comment]
Please try again using the referenced commit. It should make the namespace appear in the group again.

**Comment by kobalicek** [comment]
@doxygen Do you mean that it solves missing classes in group or just namespaces?

**Comment by doxygen** [comment]
@kobalicek as @albert-github already indicated the classes were already present in the latest version, but the namespaces had gone. That I now also repaired.

**Comment by kobalicek** [comment]
I see, sorry I just reacted to the last message :) Happy to see this fixed!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
